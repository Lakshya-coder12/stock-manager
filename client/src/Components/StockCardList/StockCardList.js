import React, { Component } from "react";
import StockCard from "../StockCard/StockCard";

class StockCardList extends Component {
  render() {
    const items = this.props.items;
    return (
      <div>
        {items.map((item) => {
          return (
            <StockCard
              key={item._id}
              name={item.name}
              companyName={item.companyName}
              stock={item.stock}
              id={item._id}
            />
          );
        })}
      </div>
    );
  }
}

export default StockCardList;
