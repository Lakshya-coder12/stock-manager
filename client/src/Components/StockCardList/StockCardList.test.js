import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import StockCardList from "./StockCardList";

configure({ adapter: new Adapter() });

describe("Testing StockCardList component", () => {
  const items = [
    {
      name: "Shoes",
      companyName: "Nike",
      stock: "100",
      _id: 1,
    },
    {
      name: "Car",
      companyName: "Mahindra",
      stock: "10",
      _id: 2,
    },
  ];
  it("matches snapshot", () => {
    const wrapper = shallow(<StockCardList items={items} />);
    expect(wrapper).toMatchSnapshot();
  });
});
