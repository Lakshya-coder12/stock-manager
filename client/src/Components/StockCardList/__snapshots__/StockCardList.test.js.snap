// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing StockCardList component matches snapshot 1`] = `
<div>
  <StockCard
    companyName="Nike"
    id={1}
    key="1"
    name="Shoes"
    stock="100"
  />
  <StockCard
    companyName="Mahindra"
    id={2}
    key="2"
    name="Car"
    stock="10"
  />
</div>
`;
